src/sentry/seer/anomaly_detection/get_historical_anomalies.py

```diff
@@ -35,26 +39,115 @@
+    )
+    data_end_time = datetime.fromtimestamp(historical_data.get(data_end_index, {}).get("timestamp"))
+    if data_end_time - data_start_time < timedelta(days=MIN_DAYS):
+        return []
+
+    # Send data to Seer
+    context = DetectHistoricalAnomaliesContext(
+        history=historical_data,
+        current=current_data,
+    )
+    body = DetectHistoricalAnomaliesRequest(
+        organization_id=organization_id,
+        project_id=project_id,
+        config=config,
+        context=context,
+    )
+    extra_data = {
+        "organization_id": organization_id,
+        "project_id": project_id,
+        "config": config,
+        "context": context,
+    }
+    try:
+        response = make_signed_seer_api_request(
+            connection_pool=seer_anomaly_detection_connection_pool,
+            path=SEER_ANOMALY_DETECTION_ENDPOINT_URL,
+            body=json.dumps(body).encode("utf-8"),
+        )
+    except (TimeoutError, MaxRetryError):
+        logger.warning("Timeout error when hitting anomaly detection endpoint", extra=extra_data)
+        return None
+
+    if response.status > 400:
+        logger.error(
+            "Error when hitting Seer detect anomalies endpoint",
+            extra={
+                "response_data": response.data,
+                "response_code": response.status,
+                **extra_data,
+            },
+        )
+        return None
+    try:
+        decoded_data = response.data.decode("utf-8")
+    except AttributeError:
+        logger.exception(
+            "Failed to parse Seer anomaly detection response",
+            extra={
+                "ad_config": config,
+                "context": context,
+                "response_data": response.data,
+                "response_code": response.status,
+            },
+        )
+        return None
+
+    try:
+        results: DetectAnomaliesResponse = json.loads(decoded_data)
+    except JSONDecodeError:
+        logger.exception(
+            "Failed to parse Seer anomaly detection response",
+            extra={
+                "ad_config": config,
+                "context": context,
+                "response_data": decoded_data,
+                "response_code": response.status,
+            },
+        )
+        return None
+
+    if not results.get("success"):
+        logger.error(
+            "Error when hitting Seer detect anomalies endpoint",
+            extra={
+                "error_message": results.get("message", ""),
+                **extra_data,
+            },
+        )
+        return None
+
+    ts = results.get("timeseries")
+    if not ts:
+        logger.warning(
+            "Seer anomaly detection response returned no potential anomalies",
+            extra={
+                "ad_config": config,
+                "context": context,
+                "response_data": results.get("message"),
+            },
+        )
+        return None
```

Comment:
i'd recommend splitting this out as another function, `handle_seer_error_responses` (or something along those lines).

Then pass the `extra_data` variable to it (rather than individual context / config stuff), and the `results`. With that, you can also standardize the loggers data a bit more as well.

```python
def handle_seer_error_responses(results, extra_data):
    def log_statement(log_level, text):
        # can also add anything from results to extra
        logger[log_level](text, extra=extra_data)
    
    if not results.get('success'):
        log_statement('error', 'Error when hitting Seer detect anomalies endpoint')
        return False
        
    # ... the other statements
    return True
```

what's nice about this is it will simplify the logger into 1 statement, and we don't need to worry about passing in the data to the log statement. It also abstracts all this code out of the main logic path, so it's easier to reason about where things are happening.

Then you can use the boolean returns to decide what to do in the get_historical method.

Reply:
I tried this out but it's having the unfortunate side effect of no longer being able to do `mock_logger.error.assert_called_with(...)` in test unless I do something like:

```
        if log_level == "error":
            logger.error(text, extra=log_data)
        elif log_level == "warning"
            ...
```

Reply:
the linter doesn't like `logger.exception` being called outside of an exception handler case either so I can't call `log_statement('exception')`

Reply:
I pushed my attempt, lmk what you think